v2/emailpassword/troubleshooting/cors-issues.mdx

```diff
@@ -0,0 +1,265 @@
+
+For example, a user is on a website that uses the domain name "example.com". This website uses a font that's hosted on a different domain name, "fonts.com". When the user's browser makes a request for the font from "fonts.com", the browser will first send a "preflight" request to "fonts.com" to check if "fonts.com" is willing to share resources with "example.com".
+
+If "fonts.com" allows cross-origin resource sharing, then the browser will proceed with loading the font. Otherwise, the browser will cancel the request.
+
+CORS is a way for web servers to say "I will accept cross-origin requests from this origin" or "I will not accept cross-origin requests from this origin". CORS is a security measure to prevent malicious scripts from making unauthorized requests.
+
+This is important because cross-origin requests can be quite scary. I could be logged into my bank account and a malicious site could make an AJAX request to the bank site to retrieve my bank balance.
+
+CORS supports these types of cross-origin requests while still maintaining a high level of security. By specifying which origin domains are allowed to make requests and which methods and headers are allowed, we can all browse the web without worrying about our bank information being stolen.
+
+
+## CORS with Supertokens
+
+CORS is an essential part to getting your Supertokens integration working!
+
+Below is a quick example of how to add CORS to work with Supertokens in an NodeJS (express) app.
+
+```tsx
+import SuperTokens from "supertokens-node";
+import express from "express";
+import cors from "cors";
+let app = express();
+
+// ...other middlewares
+
+app.use(
+  cors({
+    origin: "http://127.0.0.1:8080",
+    allowedHeaders: ["content-type", ...SuperTokens.getAllCORSHeaders()],
+    methods: ["GET", "PUT", "POST", "DELETE"],
+    credentials: true,
+  })  
+);
+
+// ...your API routes
+```
+
+## Different parts of CORS
+
+Let’s take a step back and talk about what each of these fields mean.
+
+**`origin` [must have]** -
+
+This configures the `Access-Control-Allow-Origin` CORS header. It expects a couple input types. For our case, we’ll only talk about what happens when we assign a Boolean or String to this field.
+
+- Boolean - set origin to true to reflect the request origin, as defined by req.header('Origin'), or set it to false to disable CORS.
+- String - set origin to a specific origin. For example, once set to "http://example.com" only requests from "http://example.com" will be allowed.
+
+This is important because Supertokens has both a frontend and backend SDK. The frontend SDK calls the APIs exposed by the backend SDK. As a result, if the origin to set to false or the wrong origin, we’ll find an error such as the following -
+
+:::danger
+Origin [http://localhost:3000](http://localhost:3000/) is not allowed by Access-Control-Allow-Origin. Status code: 204
+:::
```

Comment:
Don't use danger notation here - it's not the right use of this. Instead just put it in regular code snippet like this:
```
Origin http://localhost:3000 is not allowed by Access-Control-Allow-Origin. Status code: 204
```

Reply:
Fixed!